mlflow/utils/thread_utils.py

```diff
@@ -42,3 +44,4 @@ def set(self, value):
         Set a value for the thread-local variable.
         """
         self.thread_local.value = (value, os.getpid())
+        self._value_dict[threading.currentThread().ident] = value
```

Comment:
Is this change related to this PR?

Reply:
Yes. `. _value_dict` attribute is used in `_start_trace` in this PR.